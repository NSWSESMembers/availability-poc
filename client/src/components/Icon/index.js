import React from 'react';
import PropTypes from 'prop-types';

import FontAwesome from 'react-native-vector-icons/FontAwesome';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import Foundation from 'react-native-vector-icons/Foundation';
import Octicons from 'react-native-vector-icons/Octicons';

const Icon = function iconBuilder(props) {
  const iconClass = props.name.substr(0, props.name.indexOf('-'));
  const iconName = props.name.substr(props.name.indexOf('-') + 1);
  switch (iconClass) {
    case 'fa': return (<FontAwesome {...props} name={iconName} />);
    case 'mci': return (<MaterialCommunityIcons {...props} name={iconName} />);
    case 'mi': return (<MaterialIcons {...props} name={iconName} />);
    case 'fnd': return (<Foundation {...props} name={iconName} />);
    case 'oi': return (<Octicons {...props} name={iconName} />);

    default: return (<FontAwesome {...props} name={props.name} />);
  }
};
Icon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
};

export default Icon;
