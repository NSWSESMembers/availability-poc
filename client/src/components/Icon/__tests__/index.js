import React from 'react';
import renderer from 'react-test-renderer';

import Icon from '../';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Icon
      name="music"
    />,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <Icon
      name="fa-music"
    />,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
  const tree3 = renderer.create(
    <Icon
      name="mci-music"
    />,
  ).toJSON();
  expect(tree3).toMatchSnapshot();
  const tree4 = renderer.create(
    <Icon
      name="fnd-music"
    />,
  ).toJSON();
  expect(tree4).toMatchSnapshot();
  const tree5 = renderer.create(
    <Icon
      name="oi-alert"
    />,
  ).toJSON();
  expect(tree5).toMatchSnapshot();
});
