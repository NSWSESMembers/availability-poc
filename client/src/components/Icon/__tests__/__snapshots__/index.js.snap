// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders correctly 2`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "FontAwesome",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders correctly 3`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders correctly 4`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "fontcustom",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders correctly 5`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Octicons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;

exports[`renders correctly 6`] = `
<Text
  accessible={true}
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
    ]
  }
>
  
</Text>
`;
